### PR TITLE
Fix cmake to work with OE >=0.13

### PR DESCRIPTION
**Pull request by achamayou**
Explicitly include & link mbedtls, since OpenEnclave 0.13 and above no longer does it by default.

### CODE REVIEW

**Review by letmaik — APPROVED** [review]

**Comment by praenubilus** [comment]
just found this after I submit a PR. In my opinion, instead of hard coding the target library and included library, we should follow the OE suggest convention( using `OE_CRYPTO_LIB`). But either one works, currently having a project blocked by OE version issue. 
FYI
https://github.com/openenclave/openenclave-curl/pull/19

**Comment by achamayou** [comment]
@praenubilus if we did support and test the OpenSSL provider, I would agree, and I think that's where we want to go eventually, but right now the switch won't actually work.

**Comment by praenubilus** [comment]
@achamayou sounds good. I just hope this PR can be merged soon to unblock our ci/cd

**Comment by achamayou** [comment]
@praenubilus would love to do that, but the "tide" bot doesn't seem happy, I'm not sure why. I've reached out to @BRMcLaren and will follow up.

**Pull request merged**
